Fix log.cpp when compiled with `executorch.enable_et_log=false`

tarun292 opened the pull request:
Summary:
Currently, the ASR execution fails with the following error: 
`xplat/executorch/runtime/platform/log.cpp:21:25: error: unused variable 'kMaxLogMessageLength' [-Werror,-Wunused-const-variable]` (P1484525874).

This diff marks kMaxLogMessageLength as `maybe_unused`.

Differential Revision: D59641404

dbort commented on runtime/platform/log.cpp:
```diff
@@ -18,7 +18,7 @@ namespace executor {
 
 /// Maximum length of a log message.
-static constexpr size_t kMaxLogMessageLength = 256;
+__ET_UNUSED static constexpr size_t kMaxLogMessageLength = 256;
```
This isn't always unused, only when logging is disabled. Instead of annotating it here, please move the definition down into `vlogf()` inside the `#ifdef`, then it will only be defined when it's actually used. Or guard this definition with `#if ET_LOG_ENABLED` so it's only defined when it's actually used.

tarun292 replied:
Done.

tarun292 committed:
Fix log.cpp when compiled with `executorch.enable_et_log=false` (#4358)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/4358

Currently, the ASR execution fails with the following error:
`xplat/executorch/runtime/platform/log.cpp:21:25: error: unused variable 'kMaxLogMessageLength' [-Werror,-Wunused-const-variable]` (P1484525874).

This diff marks kMaxLogMessageLength as `maybe_unused`.

Reviewed By: dvorjackz

Differential Revision: D59641404
